Homework-Apr-04/Routers/Kruskal/EdgeList.cs

```diff
@@ -0,0 +1,116 @@
+namespace Kruskal;
+
+/// <summary>
+/// Represents graph via edge list sorted by edges weights.
+/// </summary>
+public class EdgeList
+{
+    private readonly List<Edge> edgeList;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="EdgeList"/> class.
+    /// </summary>
+    /// <param name="edgeList">List of edges. Will be sorted by weights.</param>
+    public EdgeList(List<Edge> edgeList)
+    {
+        this.edgeList = edgeList;
+        this.edgeList.Sort(Edge.CompareEdgesByWeight);
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="EdgeList"/> class.
+    /// </summary>
+    /// <param name="streamReader">Stream to read graph from
+    /// (graph should be represented via adjacency list,
+    /// vertices must be numbered from 1 to n).</param>
+    public EdgeList(StreamReader streamReader)
+    {
+        edgeList = new List<Edge>();
+        var linesCounter = 1;
+        for (
+            var currentLine = streamReader.ReadLine();
+            currentLine is not null;
+            currentLine = streamReader.ReadLine(),
+            ++linesCounter)
+        {
+            try
+            {
+                ParseAdjacencyListLine(currentLine);
+            }
+            catch (InvalidGraphRepresentationException exception)
```

Comment:
```suggestion
            catch (InvalidGraphRepresentationException e)
```
Обычно пишут просто e, что-то на уровне с названием i